File: SampleCode/RestApi/Clients/Java/src/main/java/io/swagger/client/model/BitRate.java
/*
 * Emby Server REST API
 * 
 */

package io.swagger.client.model;

import java.util.Objects;
import java.util.Arrays;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import io.swagger.v3.oas.annotations.media.Schema;
import java.io.IOException;
/**
 * A type for handling bit rates.      The purpose of this type is to avoid manual calculations and conversions in code,             unified handling and conversion as well as presentation through its various To\\*\\*\\*String methods.      &#x60;System.IComparable&#x60;      &#x60;System.IEquatable&#x60;1&#x60;  
 */
@Schema(description = "A type for handling bit rates.      The purpose of this type is to avoid manual calculations and conversions in code,             unified handling and conversion as well as presentation through its various To\\*\\*\\*String methods.      `System.IComparable`      `System.IEquatable`1`  ")

public class BitRate {
  @SerializedName("bps")
  private Long bps = null;

  @SerializedName("kbps")
  private Double kbps = null;

  @SerializedName("Mbps")
  private Double mbps = null;

  public BitRate bps(Long bps) {
    this.bps = bps;
    return this;
  }

   /**
   * Get bps
   * @return bps
  **/
  @Schema(description = "")
  public Long getBps() {
    return bps;
  }

  public void setBps(Long bps) {
    this.bps = bps;
  }

  public BitRate kbps(Double kbps) {
    this.kbps = kbps;
    return this;
  }

   /**
   * Get kbps
   * @return kbps
  **/
  @Schema(description = "")
  public Double getKbps() {
    return kbps;
  }

  public void setKbps(Double kbps) {
    this.kbps = kbps;
  }

  public BitRate mbps(Double mbps) {
    this.mbps = mbps;
    return this;
  }

   /**
   * Get mbps
   * @return mbps
  **/
  @Schema(description = "")
  public Double getMbps() {
    return mbps;
  }

  public void setMbps(Double mbps) {
    this.mbps = mbps;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    BitRate bitRate = (BitRate) o;
    return Objects.equals(this.bps, bitRate.bps) &&
        Objects.equals(this.kbps, bitRate.kbps) &&
        Objects.equals(this.mbps, bitRate.mbps);
  }

  @Override
  public int hashCode() {
    return Objects.hash(bps, kbps, mbps);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class BitRate {\n");
    
    sb.append("    bps: ").append(toIndentedString(bps)).append("\n");
    sb.append("    kbps: ").append(toIndentedString(kbps)).append("\n");
    sb.append("    mbps: ").append(toIndentedString(mbps)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

File: SampleCode/RestApi/Net.RestSharp/EmbyClient.Dotnet/Model/BitRate.cs
/*
 * EmbyClient.Dotnet
 */

using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using SwaggerDateConverter = EmbyClient.Dotnet.Client.SwaggerDateConverter;

namespace EmbyClient.Dotnet.Model
{
    /// <summary>
    /// A type for handling bit rates.      The purpose of this type is to avoid manual calculations and conversions in code,             unified handling and conversion as well as presentation through its various To\\*\\*\\*String methods.      &#x60;System.IComparable&#x60;      &#x60;System.IEquatable&#x60;1&#x60;  
    /// </summary>
    [DataContract]
        public partial class BitRate :  IEquatable<BitRate>
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="BitRate" /> class.
        /// </summary>
        /// <param name="bps">bps.</param>
        /// <param name="kbps">kbps.</param>
        /// <param name="mbps">mbps.</param>
        public BitRate(long? bps = default(long?), double? kbps = default(double?), double? mbps = default(double?))
        {
            this.bps = bps;
            this.kbps = kbps;
            this.Mbps = mbps;
        }
        
        /// <summary>
        /// Gets or Sets bps
        /// </summary>
        [DataMember(Name="bps", EmitDefaultValue=false)]
        public long? bps { get; set; }

        /// <summary>
        /// Gets or Sets kbps
        /// </summary>
        [DataMember(Name="kbps", EmitDefaultValue=false)]
        public double? kbps { get; set; }

        /// <summary>
        /// Gets or Sets Mbps
        /// </summary>
        [DataMember(Name="Mbps", EmitDefaultValue=false)]
        public double? Mbps { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class BitRate {\n");
            sb.Append("  bps: ").Append(bps).Append("\n");
            sb.Append("  kbps: ").Append(kbps).Append("\n");
            sb.Append("  Mbps: ").Append(Mbps).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public virtual string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="input">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object input)
        {
            return this.Equals(input as BitRate);
        }

        /// <summary>
        /// Returns true if BitRate instances are equal
        /// </summary>
        /// <param name="input">Instance of BitRate to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(BitRate input)
        {
            if (input == null)
                return false;

            return 
                (
                    this.bps == input.bps ||
                    (this.bps != null &&
                    this.bps.Equals(input.bps))
                ) && 
                (
                    this.kbps == input.kbps ||
                    (this.kbps != null &&
                    this.kbps.Equals(input.kbps))
                ) && 
                (
                    this.Mbps == input.Mbps ||
                    (this.Mbps != null &&
                    this.Mbps.Equals(input.Mbps))
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            unchecked // Overflow is fine, just wrap
            {
                int hashCode = 41;
                if (this.bps != null)
                    hashCode = hashCode * 59 + this.bps.GetHashCode();
                if (this.kbps != null)
                    hashCode = hashCode * 59 + this.kbps.GetHashCode();
                if (this.Mbps != null)
                    hashCode = hashCode * 59 + this.Mbps.GetHashCode();
                return hashCode;
            }
        }

    }
}
